api/services/S3Helper.js

```diff
@@ -0,0 +1,128 @@
+const AWS = require('aws-sdk');
+const config = require('../../config');
+
+const s3Config = config.s3;
+
+const S3_DEFAULT_MAX_KEYS = 1000;
+
+const s3Client = new AWS.S3({
+  accessKeyId: s3Config.accessKeyId,
+  secretAccessKey: s3Config.secretAccessKey,
+  region: s3Config.region,
+});
+
+function listObjectsHelper(currObjects, extraS3Params = {}, opts = {}, callback) {
+  const listObjectArgs = Object.assign({}, {
+    Bucket: s3Config.bucket,
+  }, extraS3Params);
+
+  s3Client.listObjectsV2(listObjectArgs, (err, data) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    const aggregationKey = opts.aggregationKey || 'Contents';
+    const totalMaxObjects = opts.totalMaxObjects;
+
+    const objects = currObjects ?
+      currObjects.concat(data[aggregationKey])
+      : data[aggregationKey];
+
+    // if results should be limited
+    if (totalMaxObjects) {
+      // We're ready to exit if either the response data are not truncated
```

Comment:
I think T=this behavior can be broken out into a named function, which will hopefully make CC happy